master-nonlatent/master.cfg

```diff
@@ -766,25 +765,23 @@ f_dockerlibrary.addStep(
             "-xc",
             "gh auth login --with-token < ~/gh_auth",
         ],
-        doStepIf=lambda step: (str(step.getProperty("lastsha")) != ""),
+        doStepIf=lambda step: (str(step.getProperty("lastsha")) != "null" and step.hasProperty("lastsha")),
```

Comment:
Didn't the code change above change this to `''`?

Reply:
![image](https://github.com/user-attachments/assets/6b119a4c-4868-4ab9-807e-bd7ff49fdab6)
https://buildbot.dev.mariadb.org/#/builders/4/builds/17

Same with step below.

Reply:
well,
 if last tag not found --> "" , so Determine SHA step won't run (is checking for lasttag != "")

 if last tag found --> `Determine SHA` step has the chance to return an empty tags array from QUAY API, and `jq -r .tags[0].manifest_digest'` will return `null` and in buildbot it's interpreted as the string literal `"null"` so that's why I check lastsha != "null" . I tried a different jq syntax which will return "" (For example you can write \\ "" ) but the mixing of ' and " quotes under buildbot interpolation creates a mess, I spent a few hours on quoting until I finally gave up and preferred "null"

Reply:
So in a nutshell:
 [a]  Determina SHA --> runs if last tag != "" 
 [b]  GH credentials --> runs if [a].lastsha != "null" or if [a].lastsha is defined 
 [c]  Operator test --> runs if [a].lastsha != "null" or if [a].lastsha is defined . Won't run if [b] failed.